### PR TITLE
Fix gamerule display issues

### LINKED ISSUE #27801: New event notification does not display if an admin runs the event
## Description
the log does not come up if an admin runs it. I think it should show the name of the admin or server AND actually appear when ran by an admin

**Reproduction**
Run any addgamerule command

**Screenshots**
Screenshot 2024-05-08 171723

### LINKED ISSUE #28165: The admin message about the round type should not happen until after joining admins are deadminned
There's a message about what the game mode is that I'm told goes out around 15 seconds before round start, which is probably when round type is determined and the map loads. This announcement should not happen until after admins who join the round as players are deadminned.

It sort of defeats the purpose of automatically deadminning admins who join the round if they have to manually deadmin 15 seconds before round start anyway to avoid having the round type leaked to them.

For bonus points: I don't know if this is an issue because I didn't check the channel it goes out on, but this announcement should be sent over the same channel as player joins and leaves, not the admin alerts channel, which should be used for stuff like likely rule violations or things which are likely to be mistaken for rule violations.

### PR DESCRIPTION
**Pull request by Titian3**
## About the PR
- Moved game mode notification for admins to when an admin joins the round. 
Fixes #28165
- Admin announcement when an admin manually adds a gamerule.
Fixes #27801
- New command to list gamerules mid round.

## Why / Balance
When an admin would play and be automatically de-admin'd they would still see the secret mode selection as the secret selection was made at 15s before round start to prepare map.
There was also no good/easy way to see what had been already run when coming in to a round.

## Technical details
3 changes.
- When an admin joins the game they are shown the gamerules that have been added so far for the game except for the roundstartvariation ones.
- When an admin runs the `addgamerule` command there is an admin announcement similar to how the event scheduler alerts on them but with the admin that ran it.
- Added new `listgamerules` command that will show the list of rules that have been added so far in the round.

## Media
Mainly from my testing:
- At round start for when an admin joins:
image
- When coming in later:
image
- The new command:
image
- Admin Announcement for `addgamerule` 
image
- Showing additional rules added:
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None

**Changelog**

:cl: Repo
ADMIN:
- add: Added listgamerules command, lists all run game modes for the round.
- add: Added admin anouncement for addgamerule
- fix: Gamemode now shows after starting round.

### CODE REVIEW

File: Content.Server/GameTicking/GameTicker.Player.cs [hunk 1]
```diff
@@ -196,6 +197,15 @@ public void PlayerJoinGame(ICommonSession session, bool silent = false)
             _playerGameStatuses[session.UserId] = PlayerGameStatus.JoinedGame;
             _db.AddRoundPlayers(RoundId, session.UserId);
 
+            if (_adminManager.IsAdmin(session))
```
[the remark this answers is not in the document]
Comment (Titian3) [reply Content.Server/GameTicking/GameTicker.Player.cs]: Ok have set this one now to flags, unsure if admin was high enough perm but should be ok for now.

File: Content.Server/GameTicking/GameTicker.Player.cs [hunk 2]
```diff
@@ -196,6 +197,15 @@ public void PlayerJoinGame(ICommonSession session, bool silent = false)
...
+            {
+                if (_allPreviousGameRules.Count > 0)
+                {
+                    var rulesMessage = GetGameRulesListMessage();
+                    _chatManager.SendAdminAnnouncement(Loc.GetString("starting-rule-selected-preset", ("preset", rulesMessage)));
```
[the remark this answers is not in the document]
Comment (Titian3) [reply Content.Server/GameTicking/GameTicker.Player.cs]: Updated it to new method to only admin announce/message to just the singular admin.

**Comment by moonheart08** [comment]
The in-chat table should probably use monospace font, using the appropriate tag, so it's aligned.

**Comment by moonheart08** [comment]
wRONG BUTTON

**Comment by Titian3** [comment]
> The in-chat table should probably use monospace font, using the appropriate tag, so it's aligned.

I didn't really like that it was not aligned either, I tried the ``` monospace markup tag but it didn't work. It looks like I'd have to rewrite a whole bunch of the message event and the client chatbox UI to have a stylesheet override for that UI window. Which seems a bit overkill for just this one message.
Open to change it if you know an easier way I've missed to implement it. 
For now, have a alternate solution by cleaning it up by just removing the header and break row from the table when in the chat but left it in the console where monospace is set in the stylesheet already:

image
image

**Comment by EmoGarbage404** [comment]
primo shit. good work

**Pull request merged**